logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/mlsmigration/MLSMigrator.kt

```diff
@@ -83,13 +94,32 @@ internal class MLSMigratorImpl(
 
     private suspend fun finalise(conversationId: ConversationId): Either<CoreFailure, Unit> {
         kaliumLogger.i("finalising ${conversationId.toLogString()} to mls")
-        return conversationRepository.updateProtocol(conversationId, Conversation.Protocol.MLS)
-            .flatMapLeft {
-                kaliumLogger.w("failed to migrate ${conversationId.toLogString()} to mls: $it")
+        return conversationRepository.updateProtocol(conversationId, Protocol.MLS)
+            .fold({ failure ->
+                kaliumLogger.w("failed to migrate ${conversationId.toLogString()} to mls: $failure")
                 Either.Right(Unit)
-            }
+            }, { updated ->
+                if (updated) {
+                    systemMessageBuilder.insertProtocolChangedSystemMessage(
+                        conversationProtocolEvent(conversationId, Protocol.MLS)
+                    )
+                }
+                Either.Right(Unit)
+            })
     }
 
+    private fun conversationProtocolEvent(
+        conversationId: ConversationId,
+        protocol: Protocol
+    ): Event.Conversation.ConversationProtocol =
+        Event.Conversation.ConversationProtocol(
+            id = "",
```

Comment:
Is there a particular reason why the ID is it empty ?  I see in other parts we are using a generated one, or a `LocalId.generate()`.

Also, maybe not that relevant, since it's not widely used, but this seems like a mapper class/function, so maybe that name would fit best

Reply:
The id is generated from the backend but sometimes we construct events locally and then we don't have any ID. The ID is only important during the incremental sync.

Anyway I'll refactor this to avoid having to create an event.

Reply:
Deleted this code